### DISCUSSION TITLE
How to programmatically run XUnit tests in parallel

### DISCUSSION THREAD

**Discussion by violetVo** [xunit/xunit#2319] [Question] [answered]
I have a few test collections. If I run them in VS test runner - they run parallelly.
But if I run them using `AssemblyRunner` (similar to the example here: https://github.com/xunit/samples.xunit/blob/main/TestRunner/Program.cs), the tests run sequentially.
How can I run the tests parallelly?
* I also have some assembly configurations (`[assembly: TestCollectionOrderer(...)]` and `[assembly: CollectionBehavior(MaxParallelThreads = 4)]`) - how can I apply it when I run using the `AssemblyRunner`?

Thanks,
Violet.

**Comment by bradwilson** [comment, marked as answer]
`AssemblyRunner` should already obey any settings from `xunit.runner.json` or the attributes you show above. Can you provide a repro project that shows the problem? If so, please open a new issue. Thanks!
